packages/plugin-core/src/commands/RandomNote.ts

```diff
@@ -0,0 +1,81 @@
+import { NoteProps, NoteUtils } from "@dendronhq/common-all";
+import { Uri, window } from "vscode";
+import { DENDRON_COMMANDS } from "../constants";
+import { VSCodeUtils } from "../utils";
+import { getWS } from "../workspace";
+import { BasicCommand } from "./base";
+import _ from "lodash";
+
+type CommandOpts = {};
+
+type CommandInput = {};
+
+type CommandOutput = void;
+
+export class RandomNoteCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  static key = DENDRON_COMMANDS.RANDOM_NOTE.key;
+
+  async gatherInputs(): Promise<CommandInput | undefined> {
+    return {};
+  }
+
+  async execute(_opts: CommandOpts): Promise<CommandOutput> {
+    const ws = getWS();
+    const engine = ws.getEngine();
+    const config = ws.config.randomNote;
+
+    // If no pattern is specified for include, then include all notes for the search set.
+    let includeSet: string[] = config?.include ?? [""];
+
+    let searchPredicate = function (note: NoteProps) {
+      if (note.stub == true) {
+        return false;
+      }
+
+      let isMatch = false;
+
```

Comment:
nitpick: prefer `forEach` syntax for conciseness. eg. `includeSet.forEach( pattern => { ... } )`

Reply:
Hmm - doesn't seem like you can call `break` from within an interable.forEach() loop though: https://stackoverflow.com/questions/2641347/short-circuit-array-foreach-like-calling-break